functions/snippets/src/main/java/com/example/functions/OcrProcessImage.java

```diff
@@ -0,0 +1,152 @@
+import com.google.protobuf.ByteString;
+import com.google.pubsub.v1.ProjectTopicName;
+import com.google.pubsub.v1.PubsubMessage;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.logging.Logger;
+
+public class OcrProcessImage implements BackgroundFunction<GcsEvent> {
+  // TODO<developer> set these environment variables
+  private static final String PROJECT_ID = System.getenv("GCP_PROJECT");
+  private static final String TRANSLATE_TOPIC_NAME = System.getenv("TRANSLATE_TOPIC");
+  private static final String[] TO_LANGS = System.getenv("TO_LANG").split(",");
+
+  private static final Logger LOGGER = Logger.getLogger(OcrProcessImage.class.getName());
+  private static final String LOCATION_NAME = LocationName.of(PROJECT_ID, "global").toString();
+  private static Publisher publisher;
+  private static final Gson gson = new Gson();
+
+  public OcrProcessImage() throws IOException {
+    publisher = Publisher.newBuilder(
+        ProjectTopicName.of(PROJECT_ID, TRANSLATE_TOPIC_NAME)).build();
+  }
+
+  @Override
+  public void accept(GcsEvent gcsEvent, Context context) throws RuntimeException {
+
+    // Validate parameters
+    String bucket = gcsEvent.getBucket();
+    if (bucket == null) {
+      throw new IllegalArgumentException("Missing bucket parameter");
+    }
+    String filename = gcsEvent.getName();
+    if (filename == null) {
+      throw new IllegalArgumentException("Missing name parameter");
+    }
+
+    detectText(bucket, filename);
+  }
+  // [END functions_ocr_process]
+
+  // [START functions_ocr_detect]
+  private void detectText(String bucket, String filename) throws RuntimeException {
+    LOGGER.info("Looking for text in image " + filename);
+
+    List<AnnotateImageRequest> visionRequests = new ArrayList<>();
+    String gcsPath = String.format("gs://%s/%s", bucket, filename);
+
+    ImageSource imgSource = ImageSource.newBuilder().setGcsImageUri(gcsPath).build();
+    Image img = Image.newBuilder().setSource(imgSource).build();
+
+    Feature textFeature = Feature.newBuilder().setType(Feature.Type.TEXT_DETECTION).build();
+    AnnotateImageRequest visionRequest =
+        AnnotateImageRequest.newBuilder().addFeatures(textFeature).setImage(img).build();
+    visionRequests.add(visionRequest);
+
+    // Detect text in an image using the Cloud Vision API
+    AnnotateImageResponse visionResponse;
+    try (ImageAnnotatorClient client = ImageAnnotatorClient.create()) {
+      visionResponse = client.batchAnnotateImages(visionRequests).getResponses(0);
+      if (visionResponse.hasError()) {
+        LOGGER.severe("Error detecting text: " + visionResponse.getError().getMessage());
+        return;
+      }
+    } catch (IOException e) {
+      // Cast to RuntimeException
+      throw new RuntimeException(e);
+    }
+
+    if (visionResponse == null || !visionResponse.hasFullTextAnnotation()) {
+      return;
```

Comment:
Is this a success?

Reply:
Added a log statement.

I'm reluctant to throw an exception here, as this could be caused by bad _user_ input (as opposed to _programmer_ error.)